src/networking.c

```diff
@@ -1928,25 +1923,30 @@ int _writeToClient(client *c, ssize_t *nwritten) {
     /* When the reply list is not empty, it's better to use writev to save us some
      * system calls and TCP packets. */
     if (listLength(c->reply) > 0) {
-        int ret = _writevToClient(c, nwritten);
+        int ret = _writevToClient(c);
         if (ret != C_OK) return ret;
 
         /* If there are no longer objects in the list, we expect
          * the count of reply bytes to be exactly zero. */
         if (listLength(c->reply) == 0)
             serverAssert(c->reply_bytes == 0);
     } else if (c->bufpos > 0) {
-        *nwritten = connWrite(c->conn, c->buf + c->sentlen, c->bufpos - c->sentlen);
-        if (*nwritten <= 0) return C_ERR;
-        c->sentlen += *nwritten;
+        if (server.io_uring_enabled) {
+            c->flags |= CLIENT_PENDING_WRITE_ASYNC;
+            ioUringPrepWrite(c);
+            return C_OK;
+        }
```

Comment:
mixing io_uring with synchronous calls in `_writeToClient` is a bit messy. The current change also looks incomplete to me. for instance, it doesn't handle the replication stream nor the scatter-gather IO case. I wonder if you would like to explore an io_uring version of `_writeToClient`, say `_writeToClientUsingIOUring`?